butterfree/load/writers/delta_writer.py

```diff
@@ -0,0 +1,164 @@
+        when_matched_update_condition: str = None,
+        when_matched_delete_condition: str = None,
+    ):
+        """
+        Merge a source dataframe to a Delta table.
+
+        By default, it will update when matched, and insert when
+        not matched (simple upsert).
+
+        You can change this behavior by setting:
+        - when_not_matched_insert_condition: it will only insert
+            when this specified condition is true
+        - when_matched_update_condition: it will only update when this
+            specified condition is true. You can refer to the columns
+        in the source dataframe as source.<column_name>, and the columns
+            in the target table as target.<column_name>.
+        - when_matched_delete_condition: it will add an operation to delete,
+            but only if this condition is true. Again, source and
+            target dataframe columns can be referred to respectively as
+            source.<column_name> and target.<column_name>
+        """
+        try:
+            full_table_name = DeltaWriter._get_full_table_name(table, database)
+
+            table_exists = client.conn.catalog.tableExists(full_table_name)
+            # table_is_delta = DeltaTable.isDeltaTable(client.conn, path)
+
+            if table_exists:
+                pd_df = client.conn.sql(
+                    f"DESCRIBE TABLE EXTENDED {full_table_name}"
+                ).toPandas()
+                provider = (
+                    pd_df.reset_index()
+                    .groupby(["col_name"])["data_type"]
+                    .aggregate("first")
+                    .Provider
+                )
+                table_is_delta = provider.lower() == "delta"
+
+                if not table_is_delta:
+                    DeltaWriter()._convert_to_delta(client, full_table_name)
+
+            # For schema evolution
+            client.conn.conf.set(
+                "spark.databricks.delta.schema.autoMerge.enabled", "true"
+            )
+
+            target_table = DeltaTable.forName(client.conn, full_table_name)
+            join_condition = " AND ".join(
+                [f"source.{col} = target.{col}" for col in merge_on]
+            )
+            merge_builder = target_table.alias("target").merge(
+                source_df.alias("source"), join_condition
+            )
+            if when_matched_delete_condition:
+                merge_builder = merge_builder.whenMatchedDelete(
+                    condition=when_matched_delete_condition
+                )
+
+            merge_builder.whenMatchedUpdateAll(
+                condition=when_matched_update_condition
+            ).whenNotMatchedInsertAll(
+                condition=when_not_matched_insert_condition
+            ).execute()
+        except Exception as e:
+            logger.error(f"Merge operation on {full_table_name} failed: {e}")
+
+    @staticmethod
+    def vacuum(table: str, retention_hours: int, client: SparkClient):
```

Comment:
The docstring tells us the default retention time is 7 days.

Should it be set as a default value for this parameter?

Reply:
This is Delta's default.